src/index/TextIndexReadWrite.cpp

```diff
@@ -43,7 +39,19 @@ ContextListMetaData writePostings(ad_utility::File& out,
   }
 
   meta._startScorelist = currentOffset;
-  scoreEncoder.writeToFile(out, currentOffset);
+  if (scoreIsInt) {
+    FrequencyEncode scoreEncoder(
+        postings | ql::views::transform([](const Posting& posting) {
+          return static_cast<uint16_t>(std::get<2>(posting));
+        }));
+    scoreEncoder.writeToFile(out, currentOffset);
+  } else {
+    FrequencyEncode scoreEncoder(
+        postings | ql::views::transform([](const Posting& posting) {
+          return std::get<2>(posting);
+        }));
+    scoreEncoder.writeToFile(out, currentOffset);
+  }
```

Comment:
Same comment as above: Either use a different compression for floats, or get rid of the code duplication.